From 6a55388a609d0c9f237a651a356df77d165b3e83 Mon Sep 17 00:00:00 2001
From: Robert Marshall <rmarshall@redhat.com>
Date: Tue, 14 Mar 2017 22:25:02 -0400
Subject: [PATCH 75/75] Fix setDefaultImage for s390/s390x (#1285601)

The setDefaultImage routine has never actually supported s390/s390x when
picking an arbitrary index because a zipl config never has the property
defaultIsIndex set to 1. Eliminating the extraneous check resolved the
issue on s390/s390x and does not appear to break other platforms.

Related: rhbz#1285601
---
 grubby.c | 3 +--
 1 file changed, 1 insertion(+), 2 deletions(-)

diff --git a/grubby.c b/grubby.c
index f960ebd..7219d64 100644
--- a/grubby.c
+++ b/grubby.c
@@ -2633,8 +2633,7 @@ void setDefaultImage(struct grubConfig *config, int isAddingBootEntry,
 	}
 
 	/* Case 2: user picked an arbitrary index as the default boot entry */
-	if (newDefaultBootEntryIndex >= FIRST_ENTRY_INDEX
-	    && config->cfi->defaultIsIndex) {
+	if (newDefaultBootEntryIndex >= FIRST_ENTRY_INDEX) {
 		indexToVerify = newDefaultBootEntryIndex;
 
 		/* user chose to make latest boot entry the default */
-- 
2.9.3

